### PR TITLE
UIOR-737 Fix labels for Package POL

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/common/POLFields/DetailsFields/FieldOrderFormat.js [hunk 1]
```diff
@@ -34,6 +34,13 @@ function FieldOrderFormat({
     change('cost.listUnitPriceElectronic', null);
     change('cost.listUnitPrice', null);
 
+    if (formValues.isPackage) {
+      formValues?.locations?.map((l, i) => {
+        change(`locations[${i}].quantityPhysical`, null);
+        change(`locations[${i}].quantityElectronic`, null);
+      });
```
Comment (axelhunn) [inline src/common/POLFields/DetailsFields/FieldOrderFormat.js:41]:
```suggestion
      formValues?.locations?.forEach((_, i) => {
        change(`locations[${i}].quantityPhysical`, null);
        change(`locations[${i}].quantityElectronic`, null);
      });
```

Comment (axelhunn) [reply src/common/POLFields/DetailsFields/FieldOrderFormat.js:41]: And maybe wrap it in `form.batch`?

**Review by NikitaSedyx — APPROVED** [review]

File: src/common/POLFields/FieldsLocation/FieldsLocation.js [hunk 2]
```diff
@@ -32,13 +33,19 @@ const FieldsLocation = ({
...
+  pOLineFormValues: { orderFormat, physical, eresource, isPackage } = {},
   withValidation,
 }) => {
   if (!locations) return null;
 
   const isPhysicalQuantityRequired = isLocationPhysicalQuantityRequired(orderFormat, physical?.createInventory);
   const isElectronicQuantityRequired = isLocationEresourceQuantityRequired(orderFormat, eresource?.createInventory);
+  const isPhysicalQuantityVisible = isPackage
+    ? orderFormat !== ORDER_FORMATS.electronicResource
+    : true;
```
Comment (axelhunn) [inline src/common/POLFields/FieldsLocation/FieldsLocation.js:45]:
```suggestion
  const isPhysicalQuantityVisible = !isPackage || (orderFormat !== ORDER_FORMATS.electronicResource);
```

File: src/common/POLFields/utils.js [hunk 3]
```diff
@@ -7,6 +7,6 @@ import {
 
 export const isEresource = (format) => ERESOURCES.includes(format);
 
-export const isFresource = (format) => PHRESOURCES.includes(format);
+export const isPhresource = (format) => PHRESOURCES.includes(format);
```
Comment (axelhunn) [inline src/common/POLFields/utils.js:10]:
```suggestion
export const isPhresource = PHRESOURCES.includes;
```

File: src/components/POLine/Item/ItemForm.js [hunk 4]
```diff
@@ -159,10 +159,18 @@ class ItemForm extends Component {
...
   setIsPackage = () => {
-    const isPackageValue = !this.props.formValues?.isPackage;
+    const { change, formValues } = this.props;
+    const isPackageValue = !formValues?.isPackage;
 
     this.onChangeField(isPackageValue, 'isPackage');
     this.onChangeField(isPackageValue, 'checkinItems');
+
+    if (isPackageValue) {
+      formValues?.locations?.map((l, i) => {
```
Comment (axelhunn) [inline src/components/POLine/Item/ItemForm.js:169]: Here `forEach` is more suitable too.
And again, maybe batch it?

**Review by uladislausamets — APPROVED** [review]

File: src/common/POLFields/FieldsLocation/FieldsLocation.js [hunk 5]
```diff
@@ -32,13 +33,17 @@ const FieldsLocation = ({
...
   withValidation,
 }) => {
   if (!locations) return null;
 
   const isPhysicalQuantityRequired = isLocationPhysicalQuantityRequired(orderFormat, physical?.createInventory);
   const isElectronicQuantityRequired = isLocationEresourceQuantityRequired(orderFormat, eresource?.createInventory);
+  const isPhysicalQuantityVisible = !isPackage || (orderFormat !== ORDER_FORMATS.electronicResource);
+  const isElectronicQuantityVisible = isPackage
+    ? (orderFormat === ORDER_FORMATS.electronicResource || orderFormat === ORDER_FORMATS.PEMix)
+    : true;
```
Comment (axelhunn) [inline src/common/POLFields/FieldsLocation/FieldsLocation.js:46]: I mean, every such condition could be simplified as above )

**Review by axelhunn — APPROVED** [review]

**Comment by AndreiShumski2** [comment]
@dltj Could you please update translations? Thanks

**Comment by dltj** [comment]
https://github.com/folio-org/ui-orders/pull/1159

**Pull request merged**